pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -1968,26 +1969,46 @@ var pgBuiltins = map[string]builtinDefinition{
 		},
 	),
 
-	"inet_server_addr": makeBuiltin(defProps(),
+	"inet_server_addr": makeBuiltin(
+		tree.FunctionProperties{Category: builtinconstants.CategorySystemInfo},
 		tree.Overload{
 			Types:      tree.ParamTypes{},
 			ReturnType: tree.FixedReturnType(types.INet),
-			Fn: func(_ context.Context, _ *eval.Context, args tree.Datums) (tree.Datum, error) {
-				return tree.NewDIPAddr(tree.DIPAddr{IPAddr: ipaddr.IPAddr{}}), nil
+			Fn: func(_ context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
+				if evalCtx.SQLAdvertiseAddr == "" {
+					return tree.DNull, nil
+				}
+				host, _, err := net.SplitHostPort(evalCtx.SQLAdvertiseAddr)
```

Comment:
what kind of behavior does this have if the node is behind a load balancer? (wondering if we need to think about internal addresses in CC being sensitive somehow)

Reply:
I think this info is already accessible in several places in internal tables, logs, traces. But that said, I think the pg version of this is strictly IP, not listen addr, so let me switch this to use the bound interface.

Reply:
I switched from the _advertise_ address to the _bound_ address, so it should be the IP and only the IP that gets to this function now.

Reply:
thanks for digging into it. lgtm!